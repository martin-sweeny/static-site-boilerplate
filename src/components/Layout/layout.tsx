import React from 'react'
import { useStaticQuery, graphql } from 'gatsby'
import { Global } from '@emotion/core'

import * as styles from './layout.styles'
import Header from '../Header/header'

const Layout: React.FC<Props> = ({ children }) => {
	const data = useStaticQuery(graphql`
		query SiteTitleQuery {
			site {
				siteMetadata {
					title
				}
			}
		}
	`)

	return (
		<>
			<Global {...styles.global} />
			<Header siteTitle={data.site.siteMetadata.title} />
			<styles.ContentWrapper>
				<main>{children}</main>
				<footer>
					Â© {new Date().getFullYear()}, Built with
					{` `}
					<a href="https://www.gatsbyjs.org">Gatsby</a>
				</footer>
			</styles.ContentWrapper>
		</>
	)
}

type Props = {
	children?: JSX.Element[]
}

export default Layout
