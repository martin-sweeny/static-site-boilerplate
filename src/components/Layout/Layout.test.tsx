import React from 'react'
import renderer from 'react-test-renderer'

import { PureLayout as Layout } from './Layout'

describe('Layout', () => {
	it('renders correctly', () => {
		const tree = renderer.create(<Layout title="Title" />).toJSON()
		expect(tree).toMatchSnapshot()
	})
})
