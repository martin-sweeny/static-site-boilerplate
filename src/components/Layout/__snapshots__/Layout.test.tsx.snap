// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
Array [
  .emotion-0 {
  background: rebeccapurple;
  margin-bottom: 1.45rem;
}

.emotion-0 > div {
  margin: 0 auto;
  max-width: 960px;
  padding: 1.45rem 1.0875rem;
}

.emotion-0 h1 {
  margin: 0;
}

.emotion-0 a {
  color: #fff;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<header
    className="emotion-0"
  >
    <div>
      <h1>
        <a
          href="/"
        >
          Title
        </a>
      </h1>
    </div>
  </header>,
  .emotion-0 {
  margin: 0 auto;
  max-width: 960px;
  padding: 0px 1.0875rem 1.45rem;
  padding-top: 0;
}

<div
    className="emotion-0"
  >
    <main />
    <footer>
      Â© 
      2019
      , Built with
       
      <a
        href="https://www.gatsbyjs.org"
      >
        Gatsby
      </a>
    </footer>
  </div>,
]
`;
